src/file-system.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if defined(WPE_ENABLE_FS) && WPE_ENABLE_FS
+
+#include "../include/wpe/file-system.h"
+#include <stdlib.h>
+
+static wpe_file_system_path_handler fs_handler = NULL;
+static void*                        fs_handler_data = NULL;
+
+void
+wpe_file_system_path_set_handler(wpe_file_system_path_handler handler, void* userdata)
+{
+    fs_handler = handler;
+    fs_handler_data = userdata;
+}
```

Comment:
Since this will just set the value every time is there a value in having `bool wpe_file_system_path_has_handler(void)` which would allow one to query whether one is there before setting it?

Reply:
I imagine this will be called once at most. Do we foresee that there would be the need to change the handler after the initial setup? If yes, then it may make sense to add.

There is a precedent in [wpe_audio_source_has_receiver()](https://github.com/Igalia/WPEBackend-fdo/blob/master/include/wpe/extensions/audio.h#L60). In that case a reason was that WebKit needs to know whether there is a receiver to know whether it has to handle final output to the receiver, or do it itself.

Reply:
Ok I mentioned it because you said to just overwrite what was there without a check to see if it was set. So it sounded like maybe there should be a way to check things. Not sure how things work in practice so I'll hold off on that.